sound/soc/sdca/sdca_interrupts.c

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * SDCA IRQ handler library
+ *
+ * Copyright (C) 2025 Cirrus Logic, Inc. and
+ *                         Cirrus Logic International Semiconductor Ltd.
+ *
+ * The MIPI SDCA specification is available for public downloads at
+ * https://www.mipi.org/mipi-sdca-v1-0-download
+ */
+#include <linux/device.h>
+#include <linux/soundwire/sdw.h>
+#include <linux/soundwire/sdw_registers.h>
+#include <sound/sdca.h>
+#include <linux/regmap.h>
+#include <linux/interrupt.h>
+#include <linux/workqueue.h>
+#include <sound/sdca_interrupts.h>
+#include <sound/sdca_function.h>
+
+static irqreturn_t sdca_base_irq_handler(int irq, void *data)
+{
+	struct sdca_interrupt *priv = data;
+
+	return IRQ_HANDLED;
+}
+
+#define IRQ_SDCA_OFFSET(reg_number) ((SDW_SCP_SDCA_INT##reg_number) - SDW_SCP_SDCA_INT1)
+#define IRQ_SDCA(number) REGMAP_IRQ_REG(number, \
+						    IRQ_SDCA_OFFSET((number / 8) + 1), \
```

Comment:
I don't think this will build, the string paste thing will happen before the preprocessor does the maths.